worker/provisioner/provisioner_test.go

```diff
@@ -1468,77 +1467,6 @@ func (s *ProvisionerSuite) TestHarvestAllReapsAllTheThings(c *gc.C) {
 	s.waitForRemovalMark(c, m0)
 }
 
-func (s *ProvisionerSuite) TestProvisionerRetriesTransientErrors(c *gc.C) {
```

Comment:
So, the coverage is here. The tests that we are removing is misleading.

The worker does not determine whether machine has a transient error. This is done in provisioner apiserver. All that this task is doing is calling broker.MachinesWithTransientErrors. SO we do that in the new test to return some values to test.

Machine 4 does not come u because it is not returned to the task by any of the calls.
Normal flow for machine 1 and 2 is just a provisioning that is tested elsewhere.